Created Saved Jobs Page

CMBVT2023 opened the pull request:
I was able to copy the basic design from indeed and I kept the Bay Valley Tech orange for the buttons. I also added some mock data to view what the page will look like with jobs saved to it. Any feedback or suggestions would be appreciated.

ConnorBstill commented on src/app/saved-jobs/components/ButtonWithBadge.tsx:
```diff
@@ -0,0 +1,17 @@
+            <h3 className={`${isTextBold && 'font-extrabold'} mt-auto`}>{children}</h3>
+        </Button>
+    )
+}
```
```suggestion
}

```
These red icons that Github adds means that you didn't leave a newline at the end of the file, which is something that you should do: https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log

ConnorBstill commented on src/app/saved-jobs/components/SavedJob.tsx:
```diff
@@ -0,0 +1,45 @@
+            </div>
+        </div>
+    )
+}
```
Newline here too

ConnorBstill commented on src/app/saved-jobs/placeHolderData.ts:
```diff
@@ -0,0 +1,74 @@
+    appliedJobs: jobsArr.filter(job => job.savedLocation == "Applied").length,
+    interviewsJobs: jobsArr.filter(job => job.savedLocation == "Interviews").length,
+    archivedJobs: jobsArr.filter(job => job.savedLocation == "Archived").length,
+}
```
Newline here

ConnorBstill commented on src/app/saved-jobs/placeHolderData.ts:
```diff
@@ -0,0 +1,74 @@
+]
+
+export const jobsAmounts = {
+    savedJobs: jobsArr.filter(job => job.savedLocation == "Saved").length,
```
I know this is just test data (which you implemented nicely) but we should always make it a practice to use `===` instead of `==`

ConnorBstill commented on src/components/ui/scroll-area.tsx:
```diff
@@ -0,0 +1,48 @@
+"use client"
+
+import * as React from "react"
+import * as ScrollAreaPrimitive from "@radix-ui/react-scroll-area"
```
When I try to run your branch I get this error, even after running `npm i`:
![Screenshot 2025-01-17 at 8 33 50 PM](https://github.com/user-attachments/assets/6ef3356a-8b2c-4a4e-a788-57a1342e203e)

I think shadcn usually installs these packages automatically when you add a component, maybe you forgot to commit your package.json and package-lock.json?

ConnorBstill commented on src/app/saved-jobs/page.tsx:
```diff
@@ -1,10 +1,69 @@
 export default function SavedJobsPage() {
+  const [currentTab, setCurrentTab] = useState<string>('Saved');
+  const filteredArray: Array<Job> = jobsArr.filter(
+    (job) => job.savedLocation == currentTab,
```
We should use `===` here as well. There's a few other places that you use `==`, I'm not going to comment on all of them but you should change them all to `===`

ConnorBstill commented on src/app/saved-jobs/page.tsx:
```diff
@@ -1,10 +1,69 @@
+            </ButtonWithBadge>
+          </div>
+        </div>
+       {filteredArray.length > 0 ?  
+       <ScrollArea className="container h-72 md:h-96 w-full">
+          {filteredArray.map((job) => (
+            <SavedJob
+              key={job.jobTitle}
+              jobTitle={job.jobTitle}
+              company={job.company}
+              location={job.location}
+              savedDate={job.savedDate}
+              companyImage={job.companyImage}
+            />
+          ))}
```
It's important to keep the main JSX templating as clean as possible and not include any JS logic like ternaries and maps if we can help it. Usually I move stuff like this into it's own function and would call it something like `renderSavedJobs`

ConnorBstill reviewed (changes requested):
Overall the code looks good, just a few changes. Once you commit your package.json then I can run it and see how it looks (unless I'm mistaken and my issue is caused by something else)

CMBVT2023 commented:
Thanks you for the feedback and tips. I implemented and reworked everything you recommended. I discovered that when I imported the scroll-area it for some reason did not add it to the dependencies for the project which is now fixed.

ConnorBstill reviewed (changes requested):
I can run it now and it works well and looks good, but there's two main things to change:

1. Shadcn has a Tab components that I think might look better instead of using regular buttons for the "Saved", "Applied" and other tabs. https://ui.shadcn.com/docs/components/tabs

2. The list of jobs scrolls even when it has plenty of room to show all of them. It should only scroll when there's not enough room.

CMBVT2023 commented:
I believe I fixed the scroll issue by setting a height for the main container and basing it off the user's view port. I also switched to using the tabs component. If anyone has any feedback or recommendations please let me know.

ConnorBstill commented on src/app/saved-jobs/placeHolderData.ts:
```diff
@@ -0,0 +1,127 @@
+    },
+]
+
+type JobsAmounts = {
+    [key: string]: number;
+}
```
```suggestion
interface JobsAmount {
  Saved: number;
  Applied: number;
  ...
}
```
